topicsdb/search_parallel.go

```
@@ -117,5 +118,7 @@ func (tt *index) scanPatternVariant(pos uint8, variant common.Hash, start uint64
 			break
 		}
 	}
-	onMatched(nil)
+	if _, err := onMatched(nil); err != nil {
+		log.Warn("searchParallel", "err", err)
```

since the first time i saw this line, i dont really know what to do with it. 
The search algorithm will look for something not returning an error and producing errors may be the intended use. 

The context we add to warn is not good.
- Write a complete sentence in warn:  "error produced during parallel search"
- Add a comment explaining the issue, so that  the problem is understood in case this line turns too verbose (which could affect performance of some hot region)

@HerbertJordan @kjezek do you know what this code is about?

---

This code is the implementation of the log-filter feature. You can send RPC queries to the service asking for log messages listing a given set of topics. This function is searching for those and does a call-back on `onMatched` for each element and, as it looks like, a final one passing a `nil` to indicate that all matches have been reported and that there are no more.

The only time this function is actually used is in line 94 above, where `onMatched` returns an error if, for instance, the execution context is canceled. Another option is that the consumer of the match failed to consume it, which seems to be ignored, as shown in line 116.

I guess the best option here, matching the action in line 116, is to simply ignore the error.